libs/arangodb/langchain_arangodb/chat_message_histories/arangodb.py

```diff
@@ -173,6 +173,10 @@ def add_message(self, message: BaseMessage) -> None:
             },
         )
 
+    def add_doc(self, doc: dict[str, Any]) -> None:
+        """Add a dict of message to the chat history."""
+        self._db.insert_document(self._collection_name, doc)
+
```

Comment:
Let's consider introducing a function called `add_qa_message` which is used specifically for `ArangoGraphQAChain` purposes. 

When we do this, we can the `role` property set to `"qa`, e.g:
```
{
    "_key": "...",
    "role": "qa",
    "user_input": user_input,
     "aql_query": aql_query,
      "result": result.content
}
```

Reply:
This would allow us to slightly optimize the AQL query to look like this:

```
FOR doc IN {collection_name}
    FILTER doc.role == "qa"
    SORT doc._key DESC
    LIMIT @n
    RETURN doc
```

Reply:
Idea: Maybe use `type` instead of `role`? This would require modifying the other message insertions to include a new `type` property...

Reply:
Idea: Better yet, we can introduce a `get_messages_by_role` method in `ChatHistory` that accepts `n` and `role` as the parameters